### PR TITLE
Feature/host join screen

### PR DESCRIPTION
**Pull request by lucasvdiepen**

### CODE REVIEW

**Comment by lucasvdiepen** [comment]
Some missing or dangling .meta found. Fix commits are needed.

| Status | File |
|:--|:--|
| Not removed | `Assets/Art/2D/UI/Map 2.meta` |
| Not removed | `Assets/Art/2D/UI/Map.meta` |

**Comment by lucasvdiepen** [comment]
Some missing or dangling .meta found. Fix commits are needed.

| Status | File |
|:--|:--|
| Not removed | `Assets/Art/2D/UI/Map 2.meta` |
| Not removed | `Assets/Art/2D/UI/Map.meta` |

**Comment by lucasvdiepen** [comment]
Some missing or dangling .meta found. Fix commits are needed.

| Status | File |
|:--|:--|
| Not committed | `Assets/Art/2D/UI/Hostscreen.meta` |
| Not committed | `Assets/Art/2D/UI/JoinScreen.meta` |

**Comment by lucasvdiepen** [comment]
Some missing or dangling .meta found. Fix commits are needed.

| Status | File |
|:--|:--|
| Not committed | `Assets/Art/2D/UI/Hostscreen.meta` |

**Comment by lucasvdiepen** [comment]
Some missing or dangling .meta found. Fix commits are needed.

| Status | File |
|:--|:--|
| Not committed | `Assets/Art/2D/UI/Hostscreen.meta` |

**Comment by lucasvdiepen** [comment]
Some missing or dangling .meta found. Fix commits are needed.

| Status | File |
|:--|:--|
| Not committed | `Assets/Art/2D/UI/Hostscreen.meta` |

**Comment by lucasvdiepen** [comment]
Some missing or dangling .meta found. Fix commits are needed.

| Status | File |
|:--|:--|
| Not committed | `Assets/Art/2D/UI/Hostscreen.meta` |

**Comment by lucasvdiepen** [comment]
Some missing or dangling .meta found. Fix commits are needed.

| Status | File |
|:--|:--|
| Not committed | `Assets/Art/2D/UI/Hostscreen.meta` |

File: Assets/Plugins/Fish-Networking-Discovery/NetworkDiscoveryHud.cs
Comment (PiterGroot) [inline Assets/Plugins/Fish-Networking-Discovery/NetworkDiscoveryHud.cs:1]: We dont need this network discovery HUD example file, can you remove it?

File: Assets/Scripts/Framework/Buttons/MainMenu/QuitButton.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using UnityEngine;
+
+namespace DerailedDeliveries.Framework.Buttons.MainMenu
+{
+    /// <summary>
+    /// A <see cref="BasicButton"/> class which is responsible for closing the application.
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/Buttons/MainMenu/QuitButton.cs]:
```suggestion
    /// A <see cref="BasicButton"/> class responsible for closing the application.
```
To make the summaries a bit more consistent

File: Assets/Scripts/Framework/GameManagement/GameManager.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+using FishNet.Object;
+
+using DerailedDeliveries.Framework.StateMachine.States;
+using DerailedDeliveries.Framework.Utils;
+
+namespace DerailedDeliveries.Framework.GameManagement
+{
+    /// <summary>
+    /// A class that is responsible for managing the game.
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/GameManagement/GameManager.cs]:
```suggestion
    /// A class responsible for managing the game.
```
for consistency

File: Assets/Scripts/Framework/PlayerManagement/PlayerManager.cs [hunk 3]
```diff
@@ -41,20 +44,35 @@ public class PlayerManager : NetworkAbstractSingleton<PlayerManager>
...
+
         /// <summary>
         /// The amount of players currently in the game.
         /// </summary>
         public int PlayerCount => _players.Count;
 
+        /// <summary>
+        /// The maximum amount of players allowed in the game.
+        /// </summary>
+        public int MaxPlayers => _maxPlayers;
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/PlayerManagement/PlayerManager.cs]: Change this to an auto getter, setter and serialize its field.

File: Assets/Scripts/Framework/PlayerManagement/PlayerManager.cs [hunk 4]
```diff
@@ -73,27 +91,20 @@ public bool IsSpawnEnabled
 
         private readonly List<PlayerId> _players = new();
         private readonly List<PlayerSpawnRequester> _playerSpawners = new();
+        private List<Transform> _availableSpawnpoints;
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/PlayerManagement/PlayerManager.cs:90]:
```suggestion
        
        private List<Transform> _availableSpawnpoints;
```

File: Assets/Scripts/Framework/PlayerManagement/PlayerManager.cs [hunk 5]
```diff
@@ -73,27 +91,20 @@ public bool IsSpawnEnabled
...
         private bool _isSpawnEnabled;
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/PlayerManagement/PlayerManager.cs:93]:
```suggestion
        
        private bool _isSpawnEnabled;
```

File: Assets/Scripts/Framework/PlayerManagement/PlayerManager.cs [hunk 6]
```diff
@@ -73,27 +91,20 @@ public bool IsSpawnEnabled
...
         private int _playerIdCount;
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/PlayerManagement/PlayerManager.cs:95]:
```suggestion
        
        private int _playerIdCount;
```

File: Assets/Scripts/Framework/StateMachine/StateMachine.cs [hunk 7]
```diff
@@ -399,6 +390,53 @@ private List<State> GetStateTree(State state)
             return tree;
         }
 
+        /// <summary>
+        /// Invoked when all states are initialized and the state machine is ready.
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/StateMachine/StateMachine.cs]:
```suggestion
        /// Called when all states are initialized and the state machine is ready.
```

**Review by PiterGroot — CHANGES_REQUESTED** [review]
Good stuff, don't forget to update from develop.

File: Assets/Plugins/Fish-Networking-Discovery/NetworkDiscoveryHud.cs
Comment (lucasvdiepen) [reply Assets/Plugins/Fish-Networking-Discovery/NetworkDiscoveryHud.cs:1]: Resolved

**Review by PiterGroot — APPROVED** [review]

**Pull request merged**